x-pack/plugins/task_manager/server/integration_tests/task_manager_capacity_based_claiming.test.ts

```diff
@@ -0,0 +1,331 @@
+jest.mock('../polling_lifecycle', () => {
+  const actual = jest.requireActual('../polling_lifecycle');
+  return {
+    ...actual,
+    TaskPollingLifecycle: jest.fn().mockImplementation((opts) => {
+      return new actual.TaskPollingLifecycle(opts);
+    }),
+  };
+});
+
+const { createMonitoringStats: createMonitoringStatsMock } = jest.requireMock('../monitoring');
+jest.mock('../monitoring', () => {
+  const actual = jest.requireActual('../monitoring');
+  return {
+    ...actual,
+    createMonitoringStats: jest.fn().mockImplementation((opts) => {
+      return new actual.createMonitoringStats(opts);
+    }),
+  };
+});
+
+const mockTaskTypeNormalCostRunFn = jest.fn();
+const mockCreateTaskRunnerNormalCost = jest.fn();
+const mockTaskTypeNormalCost = {
+  title: 'Normal cost task',
+  description: '',
+  cost: TaskCost.Normal,
+  stateSchemaByVersion: {
+    1: {
+      up: (state: Record<string, unknown>) => ({ foo: state.foo || '' }),
+      schema: schema.object({
+        foo: schema.string(),
+      }),
+    },
+  },
+  createTaskRunner: mockCreateTaskRunnerNormalCost.mockImplementation(() => ({
+    run: mockTaskTypeNormalCostRunFn,
+  })),
+};
+const mockTaskTypeXLCostRunFn = jest.fn();
+const mockCreateTaskRunnerXLCost = jest.fn();
+const mockTaskTypeXLCost = {
+  title: 'XL cost task',
+  description: '',
+  cost: TaskCost.ExtraLarge,
+  stateSchemaByVersion: {
+    1: {
+      up: (state: Record<string, unknown>) => ({ foo: state.foo || '' }),
+      schema: schema.object({
+        foo: schema.string(),
+      }),
+    },
+  },
+  createTaskRunner: mockCreateTaskRunnerXLCost.mockImplementation(() => ({
+    run: mockTaskTypeXLCostRunFn,
+  })),
+};
+jest.mock('../queries/task_claiming', () => {
+  const actual = jest.requireActual('../queries/task_claiming');
+  return {
+    ...actual,
+    TaskClaiming: jest.fn().mockImplementation((opts: TaskClaimingOpts) => {
+      opts.definitions.registerTaskDefinitions({
+        _normalCostType: mockTaskTypeNormalCost,
+        _xlCostType: mockTaskTypeXLCost,
+      });
+      return new actual.TaskClaiming(opts);
+    }),
+  };
+});
+
+const taskManagerStartSpy = jest.spyOn(TaskManagerPlugin.prototype, 'start');
+const alphabet = 'abcdefghijklmnopqrstuvwxyz'.split('');
+
+describe('capacity based claiming', () => {
+  const taskIdsToRemove: string[] = [];
+  let esServer: TestElasticsearchUtils;
+  let kibanaServer: TestKibanaUtils;
+  let taskManagerPlugin: TaskManagerStartContract;
+  let createMonitoringStatsOpts: CreateMonitoringStatsOpts;
+
+  beforeAll(async () => {
+    const setupResult = await setupTestServers({
+      xpack: {
+        task_manager: {
+          claim_strategy: `unsafe_mget`,
+          capacity: 10,
+          poll_interval: POLLING_INTERVAL,
+          unsafe: {
+            exclude_task_types: flatMap(alphabet, (letter) => [
+              `${letter}*`,
+              `${letter.toUpperCase()}*`,
```

Comment:
trying to exclude all task types so the only ones that run are the ones registered in this integration test. I'm prefixing the test task types with `_`.

Reply:
that's much better thanks! updated in https://github.com/elastic/kibana/pull/189431/commits/2c1ee89d9d5d5cf9e551faa4824cb3974e5703c3